### PR TITLE
Refine nshlib implementation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- nshlib: Call nsh_sysinitscript in usb or alt console like normal one
- nshlib: Call nsh_telnetstart in nsh_initialize to avoid the dupliation
- nshlib: Move commoin initialization from console_main to nsh_initialize

## Impact
nshlib

## Testing
sim:nsh

### CODE REVIEW

**Comment by masayuki2009** [comment]
@xiaoxiang781216 
Please rebase this PR to the latest master.

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 Please rebase this PR to the latest master.

Done.

File: nshlib/nsh_init.c [hunk 1]
```diff
@@ -66,25 +100,86 @@ static const struct extmatch_vtable_s g_nsh_extmatch =
...
+#ifdef CONFIG_NSH_USBDEV_TRACE
+  /* Initialize any USB tracing options that were requested */
+
+  usbtrace_enable(TRACE_BITSET);
+#endif
+
+#if defined(CONFIG_SYSTEM_NSH_SYMTAB)
+  /* Make sure that we are using our symbol table */
+
+  symdesc.symtab   = (FAR struct symtab_s *)CONFIG_SYSTEM_NSH_SYMTAB_ARRAYNAME; /* Discard 'const' */
```
Comment (pkarashchenko) [inline nshlib/nsh_init.c:135]: should a comment be on a separate line?

**Review by pkarashchenko — APPROVED** [review]

File: nshlib/nsh_init.c (hunk 1 above)
Comment (xiaoxiang781216) [reply nshlib/nsh_init.c:135]: There are many cases which append the short comment at the end, so it's better to keep as it.

**Review by hartmannathan — COMMENTED** [review]
Will nsh over telnet still work?

(I use nsh over telnet on some boards that lack serial ports.)

**Comment by hartmannathan** [comment]
> Will nsh over telnet still work?
> 
> (I use nsh over telnet on some boards that lack serial ports.)

Ah, nevermind, I see the where the call to `nsh_telnetstart` is done now. I didn't see that a moment ago.

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@hartmannathan please merge if you are fine with this changes

**Comment by hartmannathan** [comment]
> @hartmannathan please merge if you are fine with this changes

Done. Thanks!

**Pull request merged**